src/Background/interpolators.jl

```diff
@@ -44,3 +45,89 @@ function (z::ZoomInterpolator)(mesh::AbstractArray{T}) where T
     out = similar(mesh, float(T), size(mesh) .* z.factors)
     return imresize!(out, itp)
 end
+
+"""
+    IDWInterpolator(coordinates, values, weights = nothing, leafsize = 8.0)
+
+This performs Inverse Distance Weighted Interpolation
+
+`coordinates` represent the sample points filled column wise and `values` is the value at those sample points, `weight` are an additional multiplicative factor in the standard
+inverse distance weight and `leafsize` determines at what number of points to stop splitting the KDTree further.
+
+The interpolator can be called with some additional parameter being, `n_neighbors` which is the number of nearest neighbors to be considered,
+`power` is the exponent for distance in the weighing factor, `reg` is the offset for the weighing factor in denominator,
+`conf_dist` is the distance below which two points would be considered as the same point.
+
+!!! warning
+    0 < `n_neighbor` <= number of elements in tree
+
+# Examples
+```jldoctest
+julia> coordinates = [1.0 2.0 3.0 4.0 5.0 6.0 7.0 8.0 9.0 10.0; 1.0 2.0 3.0 4.0 5.0 6.0 7.0 8.0 9.0 10.0];
+
+julia> weights = [1.0 1.0 1.0 1.0 1.0 1.0 1.0 1.0 1.0 1.0];
+
+julia> value = [1.0 2.0 3.0 4.0 5.0 6.0 7.0 8.0 9.0 10.0];
+
+julia> positions = [9.9 1.0; 10.0 1.0];
+
+julia> alg = IDWInterpolator(coordinates,value, weights, 8);
+
+julia> alg(positions)
+1×2 Array{Float64,2}:
+ 9.57581  2.75894
+
+julia> IDWInterpolator(coordinates, value)(positions)
+1×2 Array{Float64,2}:
+ 9.57581  2.75894
+
+```
+"""
+struct IDWInterpolator <: BackgroundInterpolator
+    coordinates::AbstractArray
+    values::AbstractArray
+    weights::Union{AbstractArray, Nothing}
+    leafsize::Real
+
+    function IDWInterpolator(coordinates::AbstractArray, values::AbstractArray, weights::Union{AbstractArray, Nothing}, leafsize::Real)
+        leafsize <= 0 && error("Invalid leafsize=$leafsize. leafsize must be greater than 0")
+        new(coordinates, values, weights, leafsize)
+    end
+end
+
+IDWInterpolator(coordinates, value; weight = nothing, leafsize = 8) = IDWInterpolator(coordinates, value, weight, leafsize)
+
+function idw_util(alg::IDWInterpolator, idxs::AbstractArray, dist::AbstractArray, power::Real, reg::Real, conf_dist::Real)
+
+    dist[1] <= conf_dist && return alg.values[idxs[1]]
+
+    num = den = zero(eltype(dist))
+
+    for (i,j) = zip(1:length(idxs), 1:length(dist))
```

Comment:
Never assume indices start from 1, that's not necessarily true.  Use [`eachindex`](https://docs.julialang.org/en/v1/base/arrays/#Base.eachindex) instead:
```suggestion
    for (i,j) = zip(eachindex(idxs), eachindex(dist))
```